src/main/java/com/dyescape/dyescapebot/core/connectivity/BotConnection.java

```diff
@@ -0,0 +1,14 @@
+package com.dyescape.dyescapebot.core.connectivity;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+/**
+ * Generic, non-platform specific connection wrapper.
+ * @author Dennis van der Veeke
+ * @since 0.1.0
+ */
+public interface BotConnection {
```

Comment:
What purpose does this interface serve? What other platforms do we plan on connecting to?

Reply:
IRC? Minecraft? Multiple possibilities

Reply:
At the moment there's no immediate plan to support multiple platforms, but if we can already write a small layer of abstraction to make that process easier, then I'd suggest to do so from the start. None the less, it will help out with testing since it'll make things a bit easier to mock.